File: GO/Cap07/c07ex08.go
// c07ex08.go

package main
import "fmt"

func FATORIAL(N int) int {
  if N == 0 {
    return 1
  }
  return N * FATORIAL(N - 1)
}

func main() {
  var LIMITE int

  fmt.Print("Qual fatorial: ")
  fmt.Scan(&LIMITE)

  fmt.Printf("Fatorial = %d\n", FATORIAL(LIMITE))

  fmt.Println()
  fmt.Print("Tecle <Enter> para encerrar... ")
  fmt.Scanln()
}

File: SWIFT/Cap07/c07ex08.swift
// c07ex08.swift

import Foundation

func Fatorial(_ N: Int) -> Int {
    if N == 0 {
        return 1
    } else {
        return N * Fatorial(N - 1)
    }
}

var Limite: Int = 0

repeat {
    print("Qual fatorial: ", terminator: "")
    if let entraLimite = Int(readLine() ?? "") {
        Limite = entraLimite
        print("Fatorial = \(Fatorial(Limite))")
        break
    }
    else {
        print("\nEntrada invalida. Tente novamente.\n")
    }
} while (true)

print()
print("Tecle <Enter> para encerrar...", terminator: "")
_ = readLine()
